src/credentials/FabricTable.cpp

```diff
@@ -583,6 +583,11 @@ CHIP_ERROR FabricInfo::SetFabricInfo(FabricInfo & newFabric)
     ReturnErrorOnFailure(VerifyCredentials(newFabric.mNOCCert, newFabric.mICACert, newFabric.mRootCert, validContext, operationalId,
                                            fabricId, pubkey));
 
+    // Verify that public key in NOC matches public key generated by node and sent in CSRResponse message.
+    VerifyOrReturnError(newFabric.GetOperationalKey()->Pubkey().Length() == pubkey.Length(), CHIP_ERROR_INVALID_ARGUMENT);
+    VerifyOrReturnError(memcmp(newFabric.GetOperationalKey()->Pubkey().ConstBytes(), pubkey.Bytes(), pubkey.Length()) == 0,
+                        CHIP_ERROR_INVALID_ARGUMENT);
```

Comment:
Per spec, this is supposed to result in a `InvalidPublicKey` value of the `NodeOperationalCertStatus` enum.

Does that happen with these changes?  I suspect it does not ....

Reply:
You are correct. I've updated the error code returned in this case to CHIP_ERROR_INVALID_PUBLIC_KEY. This should later be correctly translated to InvalidPublicKey status code.